components/NavigationLink/index.js

```diff
@@ -0,0 +1,6 @@
+import NavigationContainer from "../NavigationContainer";
+import { StyledNavigationLink } from "./NavigationLink.style";
+
+export default function NavigationLink({ href, children }) {
+  return <StyledNavigationLink href={href}>{children}</StyledNavigationLink>;
+}
```

Comment:
Higher order components just add complexity layers to your code, as this is just renaming the component.